homeassistant/components/tedee/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Tedee sensor entities."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pytedee_async import TedeeLock
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import TedeeDescriptionEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class TedeeSensorEntityDescription(SensorEntityDescription):
+    """Describes Tedee sensor entity."""
+
+    value_fn: Callable[[TedeeLock], int | None]
+
+
+ENTITIES: tuple[TedeeSensorEntityDescription, ...] = (
+    TedeeSensorEntityDescription(
+        key="battery_sensor",
+        translation_key="battery_sensor",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement="%",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda lock: lock.battery_level,
+    ),
+    TedeeSensorEntityDescription(
+        key="pullspring_duration",
```

Comment:
What does this mean

Reply:
What the sensor does you mean? How long the lock holds the spring once it's pulled (fully unlocked, like when you turn the key completely)

Reply:
Is this settable via the API or is this a read only field?

Reply:
Actually in the current implementation you can't. That's only possible through the cloud API which is not part of the integration yet.